src/dataAcquisition/dataAcquisition.py

```diff
@@ -0,0 +1,156 @@
+import time
+import datetime
+import pandas
+from os.path import exists
+
+logPath = "../../data/publications/dataAcquisitionLog.txt"
+
+def getJsCode(path: str) -> str:
+    with open(path) as file:
+        content = file.read()
+    return content
+
+def launchFirefox(geckoPath: str) -> webdriver.Firefox: 
+    #geckodriver 0.33 added to PATH
+    #included in snap on linux distributions of firefox
+    
+    #creates Options obeject
+    options = Options()
+    #add option to launch firefox headless
+    options.add_argument("--headless=new")
+    #driver path given as Service object
+    serv = Service(geckoPath)
+
+    #creates driver object with specified options
+    driver = webdriver.Firefox(options=options, service=serv)
+    driver.set_window_size(800,600)
+    print("Headless Firefox Initialized")
+    return driver
+
+#visit website specified under URL and wait for specific element
+def navigateToWebsite(driver, URL: str,
+                      elementToWaitFor: str) -> webdriver.Firefox:    
+    dt = datetime.datetime.now()
+    dtstr = dt.strftime('%d/%m/%Y %H:%M:%S')
+    try:
+        driver.get(URL)
+    except:
+        with open(logPath,'a') as file:
+            file.write("["+dtstr+"]\tError: Page \""+URL
+                       +"\" could not be loaded.]"
+                       +type(Exception).__name_+"\n")
+        return driver
+    delay = 15 #timeout delay
+    try:
+        #wait for element to be loaded
+        myElem = WebDriverWait(driver, delay).until(
+                  EC.presence_of_element_located((By.ID, elementToWaitFor)))
+        with open(logPath,'a') as file:
+            file.write("["+dtstr+"]\tPage \""+URL+"\" loaded successfully.\n")
+    except TimeoutException:
+        with open(logPath,'a') as file:
+            file.write("["+dtstr+"]\tError: Page \""+URL
+                       +"\" loading timeout.\n")
+    except:
+        with open(logPath,'a') as file:
+            file.write("["+dtstr+"]\tError: Element \""+elementToWaitFor
+                       +"\" could not be loaded."+type(Exception).__name_+"\n")
+    finally:    
+        return driver
+
+#call to execute js code on website
+#data type of result is only determined on runtime
+#maybe typeof in js can be used to set return type
+def exejscode(driver, jscode: str):
+    dt = datetime.datetime.now()
+    dtstr = dt.strftime('%d/%m/%Y %H:%M:%S')
+    try:
+        result = driver.execute_script(jscode)
+    except:
+        with open(logPath,'a') as file:
+            file.write("["+dtstr+"]\tError while executing script. "
+                       +type(Exception).__name_+"\n")
+    finally:
+        with open(logPath,'a') as file:
+            file.write("["+dtstr+"]\tExtracted "+str(len(result))+" links.\n")
+        return result
+
+#extract names and ID from spreadsheet and create data frame with URLs
+def getNamesAndURLs(df: pandas.DataFrame) -> pandas.DataFrame:
+    names_IDs_URLs = df[["Vorname","Nachname","ID"]]
```

Comment:
```suggestion
    names_IDs_URLs = df[["firstname","lastname","WoSID","id"]]
```
id is ours, WoSID is from WebOfScience